metrics/internal/collectors/cluster-advance-feature-use.go

```diff
@@ -33,27 +128,33 @@ var _ prometheus.Collector = &ClusterAdvanceFeatureCollector{}
 func NewClusterAdvancedFeatureCollector(opts *options.Options) *ClusterAdvanceFeatureCollector {
 	client, err := rookclient.NewForConfig(opts.Kubeconfig)
 	if err != nil {
-		klog.Error(err)
+		klog.Errorf("cluster advanced feature collector failed to create client: %v", err)
 		return nil
 	}
 
-	lw := cache.NewListWatchFromClient(client.CephV1().RESTClient(), "cephclusters", metav1.NamespaceAll, fields.Everything())
-	sharedIndexInformer := cache.NewSharedIndexInformer(lw, &cephv1.CephCluster{}, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	advFeatureProviders := []AdvancedFeatureProvider{
+		NewCephClusterAdvancedFeatureProvider(client),
+		NewCephObjectStoreAdvancedFeatureProvider(client),
+		NewStorageClassAdvancedFeatureProvider(client),
```

Comment:
StorageClass would need K8s client instead of rook client, right?

Reply:
Yes, I remember you mentioned about StorageClass being a kubernetes resource. Made the changes, please take a look.